manifests/kustomize/base/installs/multi-user/pipelines-profile-controller/sync.py

```diff
@@ -98,7 +101,141 @@ def server_factory(visualization_server_image,
     """
     Returns an HTTPServer populated with Handler with customized settings
     """
+
     class Controller(BaseHTTPRequestHandler):
+        config.load_incluster_config()
```

Comment:
I suggest creating an overlay to contain all changes in this folder and `third-party/argo/base` folder. I want the community to test it out widely in various environments before we can make it default setup.

Reply:
So you mean one overlay for Openshift, vanilla Kubernetes, AWS, etc?

We would need to merge the tiny workflow-controller-configmap change, that does work with any distribution and add specific overlays in manifests/kustomize/base/installs/multi-user-artifact-isolation-{kubernetes,openshift,aws} to overwrite
manifests/kustomize/base/installs/multi-user/pipelines-profile-controller/ [sync.py, composite-controller.yaml]

If you have a better structure for the overlays please tell me.

Reply:
@chensun this structure question needs to be answered and then i can add an overlay specific to minio, S3, GCS etc.

Reply:
Hello @juliusvonkohout , the structure of overlay looks like this:

1. Create a folder called `platform-agnostic-multi-user-separate-folder` in the location of https://github.com/kubeflow/pipelines/tree/master/manifests/kustomize/env. 
2. Move the changes you made like `sync.py`, `composite-controller.yaml`, `deployment.yaml` to this new folder, and add a kustomization.yaml file to enable `kustomize build`.